OWNERS: Update SIG Release aliases

hasheddan opened the pull request:
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

(Edits from @justaugustus...)

Updates SIG Release OWNERS aliases to:

- Remove SIG Release approvers alias
  We're currently only using this for changelog approvals and those should
  really be handled by relnotes + RelEng.
- Syncs release-engineering-* with full Release Managers (https://github.com/kubernetes/sig-release/blob/master/release-managers.md#release-managers)
- Add Release Managers as build image reviewers
- Add CHANGELOG aliases
- Adds @wilsonehusin as CHANGELOG reviewer / approver for 1.21 cycle
- Add Release Managers as changelog approvers
- Add 1.21 Release Notes shadows as changelog reviewers

Includes:
- @tpepper offboarding
- @saschagrunert to SIG Chair
- @hasheddan to SIG Technical Lead

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
xref https://github.com/kubernetes/sig-release/issues/1322

**Special notes for your reviewer**:

/assign @justaugustus @saschagrunert 
/hold
/sig release
/area release-engineering
cc: @kubernetes/release-engineering 

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

hasheddan committed:
OWNERS(releng): Sync Release Managers

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>
Co-authored-by: Stephen Augustus <foo@auggie.dev>

hasheddan commented:
/area release-eng

saschagrunert reviewed (approved):
/lgtm

justaugustus reviewed (changes requested):
Add @jeremyrickard in, as he'll be confirmed as TL by EOD tomorrow.
ref: https://groups.google.com/g/kubernetes-dev/c/78E3IobBKH0

justaugustus reviewed (changes requested):
@hasheddan -- One more thing...
Can you pull in the changes from https://github.com/kubernetes/kubernetes/pull/97135?

hasheddan committed:
OWNERS(CHANGELOG): Add wilsonehusin (1.21 relnotes lead) as approver

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>
Co-authored-by: Stephen Augustus <foo@auggie.dev>

hasheddan commented:
@justaugustus I believe this is ready to go now :+1:

saschagrunert reviewed (approved):
/lgtm

justaugustus committed:
OWNERS(sig-release): Remove SIG Release approvers alias

We're currently only using this for changelog approvals and those should
really be handled by relnotes + RelEng.

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
OWNERS(sig-release): Add CHANGELOG aliases

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
OWNERS(CHANGELOG): Add release-engineering-reviewers as approvers

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
OWNERS(CHANGELOG): Add 1.21 Release Notes shadows as reviewers

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus committed:
OWNERS(build-image): Add Release Managers as reviewers

Signed-off-by: Stephen Augustus <foo@auggie.dev>

justaugustus commented:
For 1.21 relnotes shadows:
/verify-owners

justaugustus commented:
> The following users are mentioned in OWNERS file(s) but are untrusted for the following reasons. One way to make the user trusted is to add them as [members](https://git.k8s.io/community/community-membership.md#member) of the kubernetes org. You can then trigger verification by writing `/verify-owners` in a comment.
> 
>     * ashnehete
>       
>       * User is not a member of the org. Satisfy at least one of these conditions to make the user trusted.

@wilsonehusin -- Please work w/ @ashnehete on org membership.

justaugustus commented:
@hasheddan -- I tweaked some things and updated the description to reflect.

cc: @kubernetes/sig-release-leads (for review)

For CHANGELOG:
/cc @wilsonehusin

/triage accepted
/hold cancel

justaugustus commented:
Also:
/lgtm
/approve

saschagrunert reviewed (approved):
/lgtm

justaugustus commented:
For top-level approve:
/assign @liggitt @thockin @smarterclayton

xmudrii reviewed (approved):
/lgtm

cpanato reviewed (approved):
/lgtm

liggitt commented:
/approve
